documentation/developer_setup.md

```diff
@@ -1,88 +1,125 @@
+## Create Your SDK
 
+Once the skeleton for your SDK has been created, you can start writing your SDK.
+Please refer to the Azure [Go SDK API design guidelines][api_design] for detailed information on how to structure clients, their APIs, and more.
 
-### Defining Methods
+### Using Autorest
 
-Defining a method follows the format:
-```golang
-// Create creates the table with the tableName specified when NewClient was called.
-func (t *Client) Create(ctx context.Context, options *CreateTableOptions) (CreateTableResponse, error) {
-	if options == nil {
-		options = &CreateTableOptions{}
-	}
-	resp, err := t.client.Create(ctx, generated.Enum1Three0, generated.TableProperties{TableName: &t.name}, options.toGenerated(), &generated.QueryOptions{})
-	return createTableResponseFromGen(&resp), err
-}
+If your SDK doesn't require any Autorest-generated content, please skip this section.
+
+When using Autorest to generate code, it's best to create a configuration file that contains all of the parameters.
+This ensures that the build is repeatable and any changes are documented.
+The convention is to place the parameters in a file named `autorest.md`.
+Below is a template to get you started (you **must** include the yaml delimiters).
 
+```yaml
+clear-output-folder: false
+export-clients: true
+go: true
+input-file: <URI to OpenAPI spec file>
+license-header: MICROSOFT_MIT_NO_VERSION
+module: <full module name> (e.g. github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys)
+openapi-type: "data-plane"
+output-folder: <output directory>
+use: "@autorest/go@4.0.0-preview.44"
 ```
-The `(s *Client)` portion is the "receiver". Methods can be defined for either pointer (with a `*`) or receiver (without a `*`) types. Pointer receivers will not copy types on method calls and allows the method to mutate the receiving struct. Client methods should use a pointer receiver.
 
-All methods that perform I/O of any kind, sleep, or perform a significant amount of CPU-bound work must have the first parameter be of type [`context.Context`][golang_context] which allows the customer to carry a deadline, cancellation signal, and other values across API boundaries. The remaining parameters should be parameters specific to that method. The return types for methods should be first a "Response" object and second an `error` object.
+For the `use` section, the value should always be the latest version of the `@autorest/go` package.
+The latest version can be found at the NPM [page][autorest_go] for `@autorest/go`.
 
+For services that authenticate with Azure Active Directory, you **must** include the `security-scopes` parameter with the appropriate values (example below).
+
+```yaml
+security-scopes: "https://vault.azure.net/.default"
+```
```

Comment:
I like your autorest description. Might be a good idea to also have links to further autorest docs. https://github.com/Azure/autorest/blob/main/docs/generate/directives.md
https://github.com/Azure/autorest/blob/main/docs/readme.md

Reply:
Good idea, I've added this info.